packages/discord.js/src/structures/Message.js

```diff
@@ -409,11 +409,15 @@ class Message extends Base {
     }
 
     if (data.poll) {
+      if (this.poll) {
+        this.poll._patch({ ...data.poll });
+      }
+
       /**
        * The poll that was sent with the message
        * @type {?Poll}
        */
-      this.poll = new Poll(this.client, data.poll, this);
+      this.poll ??= new Poll(this.client, data.poll, this, this.channel);
```

Comment:
Maybe move this into an `else` block on the `if` above and drop the `??` to avoid evaluating `this.poll` twice in this branch